libraries/chain/controller.cpp

```diff
@@ -1359,8 +1359,25 @@ struct controller_impl {
       assert(!!legacy_root);
       assert(read_mode == db_read_mode::IRREVERSIBLE || !legacy_branch.empty());
       ilog("Transitioning to savanna, IF Genesis Block ${gb}, IF Critical Block ${cb}", ("gb", legacy_root->block_num())("cb", chain_head.block_num()));
-      auto new_root = block_state::create_if_genesis_block(*legacy_root);
-      fork_db.switch_from_legacy(new_root);
+      if (chain_head_trans_svnn_block) {
```

Comment:
Can you add some comments why this section of new code fixes restarting without blocks.log?